cmd/minishift/cmd/delete.go

```diff
@@ -46,23 +48,45 @@ func runDelete(cmd *cobra.Command, args []string) {
 
 	util.ExitIfUndefined(api, constants.MachineName)
 
+	if minishiftConfig.InstanceConfig.IsRegister {
+		if stopped, err := cluster.UnRegister(api); err != nil || stopped {
+			handleFailedUnRegistration()
+		}
+		minishiftConfig.InstanceConfig.IsRegister = false
+		minishiftConfig.InstanceConfig.Write()
+	}
+
 	fmt.Println("Deleting the Minishift VM...")
 	if err := cluster.DeleteHost(api); err != nil {
 		atexit.ExitWithMessage(1, fmt.Sprintf("Error deleting the VM: %s", err.Error()))
 	}
 
-	if err := minishiftConfig.InstanceConfig.Delete(); err != nil {
-		atexit.ExitWithMessage(1, fmt.Sprintln(fmt.Sprintf("Error deleting %s: ", minishiftConfig.InstanceConfig.FilePath), err))
+	removeInstanceConfigs()
+
+	fmt.Println("Minishift VM deleted.")
+}
+
+func handleFailedUnRegistration() {
+	userConfirmation := pkgUtil.ReadInputFromStdin("Current Minishift instance is registered with subscription, " +
```

Comment:
I changed the message a bit. Have a look at https://github.com/hferentschik/minishift/commit/ba794b476691e2b1e079fb348a4706603b01d51d

I tried to create a pull request against your feature branch, but that was confusing, since I also rebased your pull request already.